cinema/views.py

```diff
@@ -37,12 +47,30 @@ class MovieViewSet(viewsets.ModelViewSet):
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
         if self.action == "retrieve":
             return MovieDetailSerializer
-
         return MovieSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+        actors = self.request.query_params.get("actors")
+        genres = self.request.query_params.get("genres")
+        title = self.request.query_params.get("title")
+        if actors:
+            queryset = queryset.filter(
+                actors__id__in=map(int, actors.split(","))
+            )
+        if genres:
+            queryset = queryset.filter(
+                genres__id__in=map(int, genres.split(","))
+            )
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+
+        if self.action in ("list", "retrieve"):
+            return queryset.prefetch_related("actors", "genres")
+        return queryset.distinct()
```

Comment:
In the `get_queryset` method of `MovieViewSet`, the `distinct()` call is used after `prefetch_related`. While this is not incorrect, ensure that the use of `distinct()` is necessary for your use case, as it can impact performance. If `distinct()` is not needed, consider removing it.